src/Factory/Request/OrderProductFactory.php

```diff
@@ -57,21 +57,24 @@ public function create(OrderItemInterface $orderItem): OrderProduct
         /** @var ProductInterface $product */
         $product = $variant->getProduct();
 
-        $cartItem->setProductID((string) $product->getId());
-        $cartItem->setSku((string) $product->getCode());
-        $cartItem->setVariantID((string) $variant->getCode());
-        $cartItem->setVariantTitle($orderItem->getVariantName());
-        $cartItem->setTitle($orderItem->getProductName());
-        $cartItem->setQuantity($orderItem->getQuantity());
-        $cartItem->setPrice($orderItem->getTotal());
-        $cartItem->setImageUrl($this->productImageResolver->resolve($product));
-        $cartItem->setDiscount($this->getDiscount($orderItem));
-        $cartItem->setVendor($this->productAdditionalDataResolver->getVendor($product, $localeCode));
-        $cartItem->setTags($this->productAdditionalDataResolver->getTags($product, $localeCode));
-        $cartItem->setCategoryIDs($this->getCategoriesIds($orderItem));
-        $cartItem->setProductUrl($this->productUrlResolver->resolve($product, $localeCode));
-
-        return $cartItem;
+
+        $orderProduct->setProductID((string) $product->getId());
+        $orderProduct->setSku((string) $product->getCode());
+        $orderProduct->setVariantID((string) $variant->getCode());
+        $orderProduct->setVariantTitle($orderItem->getVariantName());
+        $orderProduct->setTitle($orderItem->getProductName());
+        $orderProduct->setQuantity($orderItem->getQuantity());
+        $orderProduct->setImageUrl($this->productImageResolver->resolve($product));
+        $orderProduct->setVendor($this->productAdditionalDataResolver->getVendor($product, $localeCode));
+        $orderProduct->setTags($this->productAdditionalDataResolver->getTags($product, $localeCode));
+        $orderProduct->setCategoryIDs($this->getCategoriesIds($orderItem));
+        $orderProduct->setProductUrl($this->productUrlResolver->resolve($product, $localeCode));
+
+        $discount = $this->getDiscount($orderItem);
+        $orderProduct->setPrice($orderItem->getTotal() + $discount);
+        $orderProduct->setDiscount($discount);
```

Comment:
These 3 lines above are the important changes. Rest of this is just a minor refactoring